repair/row_level.cc

```diff
@@ -2788,6 +2788,16 @@ class row_level_repair {
             auto schema_version = s->version();
             bool table_dropped = false;
 
+            auto& mem_sem = _ri.rs.memory_sem();
+            auto max = _ri.rs.max_repair_memory();
+            auto wanted = (_all_live_peer_nodes.size() + 1) * tracker::max_repair_memory_per_range();
+            wanted = std::min(max, wanted);
+            rlogger.trace("repair[{}]: Started to get memory budget, wanted={}, available={}, max_repair_memory={}",
+                    _ri.id.uuid, wanted, mem_sem.current(), max);
+            auto mem_permit = seastar::get_units(mem_sem, wanted).get0();
```

Comment:
What happens if a node wants to shut down? I guess the running repairs will stop, then this one will get the permit, and immediately stop too.

Reply:
Under what circumstances do we have two repairs at the same time? If two nodes run nodetool repair?

Reply:
ping @asias

Reply:
> What happens if a node wants to shut down? I guess the running repairs will stop, then this one will get the permit, and immediately stop too.

Yes.

>  Under what circumstances do we have two repairs at the same time? If two nodes run nodetool repair?

Yes, multiple nodetool repair requests from user is one case. Another case is that  if a single repair request contains multiple ranges, repair will start multiple ranges in parallel which trigger multiple repair instances.